frontend/app/routes/_public+/apply+/$id+/demographics-part1.tsx

```diff
@@ -27,21 +33,121 @@ export const meta: MetaFunction<typeof loader> = mergeMeta((args) => {
 export async function loader({ request, params }: LoaderFunctionArgs) {
   const bornTypes = await getLookupService().getAllBornTypes();
   const disabilityTypes = await getLookupService().getAllDisabilityTypes();
+  const { OTHER_EQUITY_TYPE_ID } = getEnv();
+  const equityTypes = await getLookupService().getAllEquityTypes();
   const applyFlow = getApplyFlow();
   const { id, state } = await applyFlow.loadState({ request, params });
-  return json({ id, state, bornTypes, disabilityTypes });
+
+  const otherEquityCode = equityTypes.find((equityType) => equityType.id === OTHER_EQUITY_TYPE_ID);
+  if (!otherEquityCode) {
+    throw new Response(`Unexpected 'Other' equity type: ${OTHER_EQUITY_TYPE_ID}`, { status: 500 });
+  }
+
+  return json({ id, bornTypes, disabilityTypes, otherEquityCode, equityTypes, state: state.demographicsPart1 });
 }
 
-export async function action({ request }: ActionFunctionArgs) {
-  return json('');
+export async function action({ request, params }: ActionFunctionArgs) {
+  const applyFlow = getApplyFlow();
+  const { id } = await applyFlow.loadState({ request, params });
+
+  const formSchema = z.object({
+    bornType: z.string({ required_error: 'empty-radio' }),
+    equity: z.string({ required_error: 'empty-radio' }),
+    otherEquity: z.string().min(1, { message: 'empty-field' }).optional(),
+    disabilityType: z.string({ required_error: 'empty-radio' }),
+  });
```

Comment:
If we look at the "CDCP - Decision log" > "DEV - Apply Online" sheet > (currently) Item 20, there is this question posed: "Now that someone can skip the Demographic survey, does it mean that if they do want to do it, all the fields are mandatory?".

The response is basically no validation is required except for those fields with the "Other" (Questions 2, 3, and 5) meaning the validation for `bornType`, `equity`, and `disabilityType` aren't needed.

I won't comment on whether I agree with that requirement.

Reply:
Thanks for the update. Trimmed in commit 6a3a0642e2595192055662987827b99225e6213b.